src/zenml/config/step_configurations.py

```diff
@@ -187,3 +194,48 @@ class Step(StrictBaseModel):
 
     spec: StepSpec
     config: StepConfiguration
+
+    def encode(self) -> bytes:
+        """Encodes the step into a byte string.
+
+        Returns:
+            The encoded step.
+        """
+        step_dict = self.dict(exclude_none=True)
+        step_json = json.dumps(step_dict, sort_keys=True)
+        encoded = zlib.compress(step_json.encode())
```

Comment:
Wonder whether it's worth being explicit about the level of compression here? Not quite sure where we are on the scale of speed:compression, but I don't think it's bang in the middle... https://docs.python.org/3/library/zlib.html#zlib.compress

Reply:
Removed encoding again according to reviewer consensus